themes/bootstrap3/templates/search/controls/showing.phtml

```diff
@@ -0,0 +1,30 @@
+<?
+  $transParams = [
+    '%%start%%' => $this->localizedNumber($this->results->getStartRecord()),
+    '%%end%%'   => $this->localizedNumber($this->results->getEndRecord()),
+    '%%total%%' => $this->localizedNumber($this->recordTotal),
+    '%%lookfor%%' => $this->escapeHtml($this->lookfor)
+  ];
+  $showingResults = '';
+?>
+<? if (!isset($this->skipTotalCount)): ?>
+  <? $showingResults = $this->translate('showing_results_of_html', $transParams); ?>
+<? else: ?>
+  <? $showingResults = $this->translate('showing_results_html', $transParams); ?>
+<? endif; ?>
+<? if (isset($this->overrideSearchHeading)): ?>
+  <? echo $this->layout()->srmessage = $showingResults; ?>
+  <?=$this->overrideSearchHeading?>
+<? elseif ($this->params->getSearchType() == 'basic'): ?>
+  <? if (!isset($this->skipTotalCount)): ?>
+    <? $showingResults = $this->translate('showing_results_of_for_html', $transParams); ?>
+  <? else: ?>
+    <? $showingResults = $this->translate('showing_results_for_html', $transParams); ?>
+  <? endif; ?>
+  <? $this->layout()->srmessage = $showingResults; ?>
+<? endif; ?>
```

Comment:
I agree with the idea of creating a separate template to avoid redundancy, but I still think the logic in here has gotten a bit muddled.

I'm not sure what the point of line 16 is -- if overrideSearchHeading is set, I think that's supposed to override the default search heading, so I don't know why we are assigning the overridden value to the screen reader message and then echoing it as well. I think if anything we should be sending the override value to the screen reader area as well as the screen.

Also, lines 19-23 appear completely unnecessary, since they duplicate lines 10-14, which should have already executed by the time the code reaches that point.